internal/client/client.go

```diff
@@ -0,0 +1,3 @@
+package client
+
+//go:generate mockgen -package=client -destination mock_client.go sigs.k8s.io/controller-runtime/pkg/client Client
```

Comment:
Will `go generate` see it if it is in a test file?

Reply:
I'm rather happy with having a simple file with that comment. Maybe putting it into an `internal/testutil/client` package would be a bit more explicit though.

Reply:
After discussing it offline, we'll leave it as it is for the moment. Because it's an `internal` file it shouldn't be imported. We might re-consider this in the future.